batcher/aligned-sdk/src/core/types.rs

```diff
@@ -89,9 +89,36 @@ impl NoncedVerificationData {
 // Defines an estimate price preference for the user.
 #[derive(Debug, Serialize, Deserialize, Clone)]
 pub enum PriceEstimate {
-    Min,
     Default,
     Instant,
+    Custom(usize),
+}
+
+impl TryFrom<String> for PriceEstimate {
+    type Error = String;
+
+    fn try_from(value: String) -> Result<Self, Self::Error> {
+        let val = match value.as_str() {
+            "default" => Self::Default,
+            "instant" => Self::Instant,
+            s if s.starts_with("custom") => {
+                let n = s
+                    .split_whitespace()
+                    .nth(1)
+                    .ok_or("Failed to Parse: `number_proofs_per_batch` not supplied, correct usage \"custom <NUM_PROOFS_IN_BATCH>\" (note quotations)")?
+                    .parse()
+                    .map_err(|_| "Failed to Parse: Value of `number_proofs_per_batch` invalid , correct usage \"custom <NUM_PROOFS_IN_BATCH>\" (note quotations)")?;
+                PriceEstimate::Custom(n)
+            }
+            _ => {
+                return Err(
+                    "Invalid network, possible values are: \"default\", \"instant\", \"custom <NUM_PROOFS_IN_BATCH>\""
```

Comment:
I think this message is misleading, it's not the network that is invalid, it's the estimation method.
Also, clap allows creating mutually exclusive flags, so you may have these instead:
```
--default-fee-estimate
--instant-fee-estimate
--custom-fee-estimate <NUM_PROOFS_IN_BATCH>
```
Making these three mutually exclusive means the parameter for the custom estimate can be handled by clap rather than expecting the user to pass  its argument in quotes so it is a single string.

Reply:
Here you have an example in the first answer:
https://users.rust-lang.org/t/clap-derive-required-flags-or-options-that-are-exclusive-to-each-other/75998/2

Reply:
Thanks @Oppen addressed the incorrect msg. Will change this.

Reply:
Mutual exclusive flags were added. Issue made https://github.com/yetanotherco/aligned_layer/issues/1585

Reply:
Nice!